### ISSUE TITLE
no-var should not fix variables named 'let'

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#11830]
**Tell us about your environment**

* **ESLint Version:** 5.16.0
* **Node Version:** 10.16.0
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parserOptions: {
    ecmaVersion: 6,
  },
  rules: {
    "no-var": "error",
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
// this is valid code in non-strict mode

function foo() {
  var let;
}

function bar() {
  var { let } = {};
}

```

```bash
eslint index.js --fix
```

**What did you expect to happen?**

No autofix, because it produces SyntaxError.

**What actually happened? Please include the actual, raw output from ESLint.**

```js
function foo() {
  let let; // SyntaxError
}

function bar() {
  let { let } = {}; // SyntaxError
}
```

Off topic: ESLint parser doesn't treat `let { let } = {}` as SyntaxError (probably an Acorn bug)

**Are you willing to submit a pull request to fix this bug?**
yes

**Comment by g-plane** [comment]
Hey @mdjermanovic , thanks for the issue.

However, if you enable this `no-var` rule, you're intending to target an ES6 environment. If you're going to run your code at ES5 environment, you should disable this rule.

**Comment by mdjermanovic** [comment]
@g-plane 

The example is still a valid code in ES6 and beyond in non-strict mode, 

`let` is a valid identifier. It isn't a reserved word, it's just explicitly forbidden in `let` and `const` declarations.

**Comment by mysticatea** [comment]
I confirmed it in the spec: https://www.ecma-international.org/ecma-262/9.0/#sec-let-and-const-declarations-static-semantics-early-errors

**Comment by kaicataldo** [comment]
> Off topic: ESLint parser doesn't treat let { let } = {} as SyntaxError (probably an Acorn bug)

@mdjermanovic Would you mind reporting that to https://github.com/acornjs/acorn so that it's tracked somewhere?

**Comment by mdjermanovic** [comment]
Sure, reported in [acorn #837](https://github.com/acornjs/acorn/issues/837) now.
